pkg/util/logging/default_logger.go

```diff
@@ -38,6 +38,23 @@ func DefaultLogger(level logrus.Level, format Format) *logrus.Logger {
 
 	if format == FormatJSON {
 		logger.Formatter = new(logrus.JSONFormatter)
+		// Error hooks inject nested fields under "error.*" with the error
+		// string message at "error".
+		//
+		// This structure is incompatible with recent Elasticsearch versions
+		// where dots in field names are automatically expanded to objects;
+		// field "error" cannot be both a string and an object at the same
+		// time.
+		//
+		// ELK being a popular choice for log ingestion and a common reason
+		// for enabling JSON logging in the first place, we avoid this mapping
+		// problem by nesting the error's message at "error.message".
+		//
+		// This also follows the Elastic Common Schema (ECS) recommendation.
+		// https://www.elastic.co/guide/en/ecs/current/ecs-error.html
+		logrus.ErrorKey = "error.message"
```

Comment:
For the benefit of other reviewers.
Using `--log-format=json`
Current log messages are like:
```json
{
  "backupLocation": "default",
  "controller": "backup-sync",
  "error": "rpc error: code = Unknown desc = RequestError: send request failed\ncaused by: Get https://192.168.0.100/velero?delimiter=%2F&list-type=2&prefix=backups%2F: x509: certificate signed by unknown authority",
  "error.file": "/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/object_store.go:331",
  "error.function": "main.(*ObjectStore).ListCommonPrefixes",
  "level": "error",
  "logSource": "pkg/controller/backup_sync_controller.go:175",
  "msg": "Error listing backups in backup store",
  "time": "2020-08-17T22:42:00Z"
}
```
The changes in this PR proposes this to be like
```
{
  "backupLocation": "default",
  "controller": "backup-sync",
  "error.file": "/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/object_store.go:331",
  "error.function": "main.(*ObjectStore).ListCommonPrefixes",
  "error.message": "rpc error: code = Unknown desc = RequestError: send request failed\ncaused by: Get https://192.168.0.100/velero?delimiter=%2F&list-type=2&prefix=backups%2F: x509: certificate signed by unknown authority",
  "level": "error",
  "logSource": "pkg/controller/backup_sync_controller.go:175",
  "msg": "Error listing backups in backup store",
  "time": "2020-08-17T22:40:34Z"
}
```

This is a reasonable change and I am OK with this.